integration-mc1-15/src/main/kotlin/org/eln2/Eln2.kt

```diff
@@ -25,17 +30,30 @@ object Eln2 {
      */
     fun initialize() {
         // Register blocks and items
-        ModItems.values().forEach { registry.registerItem(it.items, it.name.toLowerCase()) }
-        ModBlocks.values().forEach { registry.registerBlock(it.block, it.name.toLowerCase()) }
-		registry.initialize(thedarkcolour.kotlinforforge.forge.MOD_BUS)
-	}
+        registry.initialize(thedarkcolour.kotlinforforge.forge.MOD_BUS)
+    }
 
     /**
      * Preinit handler.
      *
      * This is run in a threaded context, so we cannot communicate with other mods without using IMC.
      */
-    fun setup(event: FMLCommonSetupEvent) {
-        OreGen.setupOreGeneration()
+
+    fun blockRegistry(event: RegistryEvent.Register<Block>){
+        ModBlocks.values().forEach {
+            registry.registerBlock(it.block, it.name.toLowerCase())
+        }
+    }
+
+    fun itemRegistry(event: RegistryEvent.Register<Item>){
+        ModItems.values().forEach {
+            registry.registerItem(it.items, it.name.toLowerCase())
+        }
+    }
```

Comment:
Is there any benefit to splitting this out into functions? I don't think there is.

Also, I want to avoid using KFF, because I'm on the fence of even including it, rather than shadowing in Kotlin ourselves. Because it doesn't even show up as a mod dependency in Minecraft, I feel that it is non-user intuitive and not a good thing for us to use. We're already going to have to shadow our application jars, and the textures and stuff are going to make the jars pretty large anyways.

Reply:
Hmm. If these functions are called by annotations, it's very unclear that is happening to me.